Use guix-provided qjs instead of a meson subproject.

Patch by bdunahu <bdunahu@operationnull.com>.

diff --git a/libr/lang/meson.build b/libr/lang/meson.build
index 0742e0e531..a6367cea39 100644
--- a/libr/lang/meson.build
+++ b/libr/lang/meson.build
@@ -4,13 +4,11 @@ r_lang_sources = [

 spp_inc = [platform_inc, include_directories('../../shlr/spp')]

-qjs = subproject('qjs')
-qjs_inc = qjs.get_variable('incs')
-qjs_static = qjs.get_variable('qjs_static')

+qjs_dep = cc.find_library('qjs', required: true)

 r_lang = library('r_lang', r_lang_sources,
-  include_directories: [platform_inc, spp_inc, qjs_inc],
+  include_directories: [platform_inc, spp_inc],
   c_args: library_cflags,
   dependencies: [
     r_util_dep,
@@ -21,19 +19,23 @@ r_lang = library('r_lang', r_lang_sources,
     r_esil_dep,
     r_muta_dep,
     r_cons_dep,
+    qjs_dep,
   ],
-  link_with: qjs_static,
   install: true,
   implicit_include_directories: false,
   install_rpath: rpath_lib,
   soversion: r2_libversion
 )

-r_lang_dep = declare_dependency(link_with: r_lang, include_directories: [platform_inc, qjs_inc])
+r_lang_dep = declare_dependency(
+  link_with: r_lang,
+  include_directories: [platform_inc],
+  dependencies: [qjs_dep],
+)

 if get_option('blob')
 r_lang_static = static_library('r_lang_static', r_lang_sources,
-  include_directories: [platform_inc, spp_inc, qjs_inc],
+  # include_directories: [platform_inc, spp_inc qjs_inc], # unused
   c_args: library_cflags,
   dependencies: [
     r_util_static_dep,
@@ -43,13 +45,15 @@ r_lang_static = static_library('r_lang_static', r_lang_sources,
     r_muta_static_dep,
     r_cons_static_dep,
   ],
-  link_with: qjs_static,
+  # link_with: qjs_static, # unused
   install: true,
   implicit_include_directories: false,
 )

-r_lang_static_dep = declare_dependency(link_with: r_lang_static,
-  include_directories: [platform_inc, qjs_inc])
+r_lang_static_dep = declare_dependency(
+  link_with: r_lang_static,
+  # include_directories: [platform_inc, qjs_inc] # unused
+  )
 endif

 pkgconfig_mod.generate(r_lang,
